pkg/csi_driver/controller_test.go

```diff
@@ -907,3 +907,73 @@ func TestCreateSnapshot(t *testing.T) {
 		}
 	}
 }
+
+func TestGetCloudInstancesReservedIPRanges(t *testing.T) {
+	cases := []struct {
+		name                       string
+		initMultishareInstanceList []*file.MultishareInstance
+		instance                   *file.ServiceInstance
+		expectIPRange              map[string]bool
+		expectErr                  bool
+	}{
+		{
+			name: "existing instances in different vpc networks",
+			initMultishareInstanceList: []*file.MultishareInstance{
+				{
+					Name:     "test-instance",
+					Project:  testProject,
+					Location: "us-west1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: testInstanceScPrefix,
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/24",
+						ConnectMode:     directPeering,
+						Name:            testVPCNetwork,
+						Ip:              "10.0.0.2",
+					},
+					Tier: enterpriseTier,
+				},
+				{
+					Name:     "test-instance-1",
+					Project:  testProject,
+					Location: "us-west1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: testInstanceScPrefix,
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.1.1.0/24",
+						ConnectMode:     directPeering,
+						Name:            defaultNetwork,
+						Ip:              "10.1.1.2",
+					},
+					Tier: enterpriseTier,
+				},
+			},
+			instance: &file.ServiceInstance{
+				Project:  testProject,
+				Name:     testCSIVolume,
+				Location: testLocation,
+				Tier:     defaultTier,
+				Network: file.Network{
+					Name:        defaultNetwork,
+					ConnectMode: directPeering,
+				},
+			},
+			expectIPRange: map[string]bool{"192.168.92.32/29": true, "192.168.92.40/29": true, "10.1.1.0/24": true},
```

Comment:
why would there be `192.168.92.32/29` and `192.168.92.40/29`?

Reply:
[Here's](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/cloud_provider/file/fake.go#L119-L143) where they are defined. A fake `ListInstances` will be called from the fake controller during the test.